### PR TITLE
Auto Update Error Handlng Fix

### PR DESCRIPTION
**Pull request by niteskum**

### CODE REVIEW

**Comment by niteskum** [comment]
ping @nethip @vickramdhawal  for review

**Comment by nitheshgangadharsalian** [comment]
Bug: https://github.com/adobe/brackets/issues/14411

**Comment by nethip** [comment]
@niteskum Could you provide description here as to how the bugs has been fixed?

File: src/extensions/default/AutoUpdate/main.js [hunk 1]
```diff
@@ -321,15 +322,14 @@ define(function (require, exports, module) {
...
+        updateDomain.on('data', receiveMessageFromNode);
 
         updateDomain.exec('initNode', {
             messageIds: MessageIds,
             updateDir: updateDir,
             requester: domainID
         });
 
-        updateDomain.on('data', receiveMessageFromNode);
-        initState();
```
Comment (nethip) [inline src/extensions/default/AutoUpdate/main.js]: `initState()` not required anymore?

**Comment by niteskum** [comment]
@nethip  Since initNode() and Brackets Side initState()  function were called one after another.  So there were no synchronization between setUpdateInProgressState() and CheckUpdateStatus()
InitNode Function notifies Brackets to reset inProgress Flag when Bracket is launched. and initState() function used to call checkUpdateFunction() which cleans up Update Temp Folder If Update was initiated in prev session. sometime UpdateTemp were being cleaned up before setting up the flags.  and Exception were thrown in setUpdateInProgress() function because there was no File to write.

Changes in this PR: 
Renamed the function "setAutoUpdateInProgressFlag" to "nodeDomainInitialized". and from initNode Function Node Notifies to Brackets to execute "nodeDomainInitialized" function and pass a parameter "reset" which is true If Brackets is launched. and false If New Brackets Window is opened.
and in nodeDomainInitialized function , initState() Function is called . and resetting of "InProgress"Flag and  checkUpdateStatus() are executed in sequence.
so this makes sure There is No RaceCondtion between resetting the flag and cleaning up the temp folder.

File: src/extensions/default/AutoUpdate/main.js (hunk 1 above)
Comment (niteskum) [reply src/extensions/default/AutoUpdate/main.js]: Now initState() is called from nodeDomainInitialized() function. so it is not required  here.

File: src/extensions/default/AutoUpdate/main.js [hunk 2]
```diff
@@ -594,11 +594,17 @@ define(function (require, exports, module) {
...
+    function nodeDomainInitialized(reset) {
+        if(reset) {
+            updateJsonHandler.parse()
+                .done(function() {
+                    setUpdateStateInJSON("autoUpdateInProgress", !reset)
+                        .always(initState);
+                 })
+                 .fail(initState);
+        } else {
+            initState();
```
[the remark this answers is not in the document]
Comment (niteskum) [reply src/extensions/default/AutoUpdate/main.js]: Restructured the Code. , Removed the CheckUpdateStatus() from initNode() function. Now initNode Function returns Promise . which is resolved when parsing is done. initNode is called and in done function checkUpdateStatus() function is called.

**Review by nethip — APPROVED** [review]

**Pull request merged**